src/main/java/uk/gov/hmcts/reform/managecase/client/datastore/StartEventResource.java

```diff
@@ -2,13 +2,17 @@
 
 import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
 import com.fasterxml.jackson.annotation.JsonProperty;
-import lombok.Data;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Getter;
 
-@Data
+@Getter
+@Builder
+@AllArgsConstructor
 @JsonIgnoreProperties(ignoreUnknown = true)
 public class StartEventResource {
     @JsonProperty("case_details")
-    private CaseDetails caseDetails;
+    private CallbackCaseDetails caseDetails;
```

Comment:
If we really need two forms of CaseDetails classes: can we use a different name as not all cases use callbacks.